### ISSUE TITLE
Where is the original source for lib/es5.js, lib/split.js?

### ISSUE THREAD

**Issue by junaruga** [lautis/uglifier#99]
Hi,
Could you tell me where is the original source or repository for `lib/es5.js`, `lib/split.js`?
For example, when we want to update lib/es5.js, lib/split.js, how do we do?
I want to know it, and also LICENSE information for those files.

Thanks.

**Comment by lautis** [comment]
`split.js` is from https://gist.github.com/slevithan/2048056 and `es5.js` snippets from Mozilla Developer Network. There is attribution and information about MIT in `split.js`. Code samples from MDN are licensed depending on their time of authoring either public domain or MIT.

Both files are only relevant if the JS engine does not support ES5. So far I've only heard of one: Windows Script Host.

**Comment by junaruga** [comment]
@lautis I got it. Thanks for your info
Let me check the original source files.
If we could include the logic to update those JavaScript files from the upstream to "rake js", I might be happy.

**Comment by lautis** [comment]
That would be cool if it can be achieved without adding complexity to the build.

**Comment by junaruga** [comment]
I think that it is easy to implement for `split.js`.
But it is tricky to implement for `es5.js`.

I could not find "git repository" for the MDN JavaScript code.

The methods in es5.js, have common structure in the web page.
And it is good to do web scraping.

- Array#forEach
- Array#map
- Array#reduce
- Array#indexOf
- Object#keys
- Object#create
- Array#filter
- Function#bind
- Array#isArray
- String#trim

The common URL formula and page structure is like this.

## URL
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/{class_name}/{method_name}

## Page

```
<h2 id="Polyfill">Polyfill</h2>
...
<pre class="brush: js">
  => The code which we want to get, is here.
</pre>
```

ex. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Only Object#defineProperty is special. It may be included in a patch file.

How do you think about this idea about es5.js ?

However, I actually I have removed split.js and es5.js in my managed RPM package, because those are used for JScript. Linux environment does not be affected for it.
So, right now I do not have a motivation to implement this for es5.js. Haha. Sorry.

**Comment by junaruga** [comment]
Hi,
Now I am trying to ask split.js's owner to merge my patch for us to use split.js without modification, instead of https://github.com/lautis/uglifier/pull/101
https://gist.github.com/slevithan/2048056

**Comment by lautis** [comment]
Fixed in #101.
